### PR TITLE
[RISCV] Remove {s,u}int_to_fp custom op action for f16/bf16

### PR DESCRIPTION
**Pull request by lukel97**
It turns out that {s,u}int_to_fp nodes get their operation action from their operand's type, not the result type, so we don't need to set it for fp16 or bf16. vp_{s,u}int_to_fp uses the result type though so we need to keep it.

This also means that we can lower int_to_fp for fixed length bf16 vectors already, so this adds tests for that.

The cost model test changes are due to BasicTTIImpl's getCastInstrCost not taking into account that int_to_fp needs its legal type swapped. This can be fixed in a later patch, but its worth noting that the affected types in the tests currently crash when lowered anyway (due to them needing split at LMUL > 8)

### CODE REVIEW

**Review by topperc — APPROVED** [review]
LGTM

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 1]
```diff
@@ -1071,9 +1071,7 @@ RISCVTargetLowering::RISCVTargetLowering(const TargetMachine &TM,
       setOperationAction({ISD::VP_MERGE, ISD::VP_SELECT, ISD::SELECT}, VT,
                          Custom);
       setOperationAction(ISD::SELECT_CC, VT, Expand);
-      setOperationAction({ISD::SINT_TO_FP, ISD::UINT_TO_FP, ISD::VP_SINT_TO_FP,
-                          ISD::VP_UINT_TO_FP},
-                         VT, Custom);
+      setOperationAction({ISD::VP_SINT_TO_FP, ISD::VP_UINT_TO_FP}, VT, Custom);
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVISelLowering.cpp:1074]: Does this mean the VP intrinsics use a different operand than the non-VP for this?

Comment (lukel97) [reply llvm/lib/Target/RISCV/RISCVISelLowering.cpp:1074]: Yeah, their "legal pos" is always the result type. We could change it to match

**Comment by lukel97** [comment]
Looks like the cost model tests are failing, so this isn't as NFC as I thought. BasicTTIImpl's getCastInstrCost isn't taking into account the swapped "legal" type either.

Since the affected test is for a type that will crash when lowering because it's > LMUL 8 and we don't split fixed length f16/bf16 vectors, I'm not too worried that the cost of it has increased.

**Review by wangpc-pp — APPROVED** [review]
LGTM.

**Comment by topperc** [comment]
> Since the affected test is for a type that will crash when lowering because it's > LMUL 8 and we don't split fixed length f16/bf16 vectors, I'm not too worried that the cost of it has increased.

Doesn't the type legalizer split them? Or do you mean we don't split before promoting?

**Comment by lukel97** [comment]
> > Since the affected test is for a type that will crash when lowering because it's > LMUL 8 and we don't split fixed length f16/bf16 vectors, I'm not too worried that the cost of it has increased.
> 
> Doesn't the type legalizer split them? Or do you mean we don't split before promoting?

The type legalizer splits them, but when we do our own promotion for {s,u}int_to_fp via custom lowering we only check for scalable vector types in `isPromotedOpNeedingSplit`

**Pull request merged**